### DISCUSSION TITLE
How to clone a Model?

### DISCUSSION THREAD

**Discussion by bymoe** [mobxjs/mobx#2702] [General] [answered]
Let's say I have a simple Model with observable `id`.

```
class TodoStore {
  todos = [];
  id: generateId();


  get completedTodosCount() {
    return this.todos.filter(
      todo => todo.completed === true
    ).length;
  }
}

const todoStore = new TodoStore();
```

now I want to add a feature to clone this todo just by duplicate this exact model with new generated id with it's todos and any extra infos.

How to do that? `.toJS` and `.createViewModel` doesn't seem to work!

**Comment by kubk** [comment, marked as answer]
It's up to you how to implement the clone method according to your business rules:

```js
class TodoStore {
  ...
  
  clone() {
    const clone = new TodoStore();
    clone.todos = this.todos; // Or use Object.assign here
    return clone;
  }
}
```

The logic of duplication might be different. For example, I could use autoincrement ID instead of UUID:

```js
class TodoStore {
  ...
  
  clone() {
    const clone = new TodoStore(this.id + 1);
    clone.todos = this.todos;
    return clone;
  }
}
```

**Reply by bymoe** [reply]
Thank you so much @kubk,
I just misunderstood how classes work. everything's clear now!
